### DISCUSSION TITLE
关于文档和热心博客们都没有提到的特殊文章内容高亮部分是如何实现的?

### DISCUSSION THREAD

**Discussion by JIeJaitt** [ppoffice/hexo-theme-icarus#1187] [Q&A] [answered]
想知道 @ppoffice 大佬究竟是如何实现这种效果的

## 需要实现的结果
<img width="651" alt="image" src="https://user-images.githubusercontent.com/77219045/219441359-1e7ea8c3-561f-429a-8d2e-df33bee5e60e.png">

<img width="1037" alt="image" src="https://user-images.githubusercontent.com/77219045/219441536-58f69737-0f69-4d00-8961-1dd5d61e6373.png">

## 网上提到的解决办法
```html
<article class="message message-immersive is-warning">
  <div class="message-body">
    <i class="fas fa-question-circle mr-2"></i>文章内容有误？请点击 
    <a href="你自己接收反馈的URL">此处</a>提交修改。
  </div>
</article>
```

```html
<article class="message message-immersive is-primary">
  <div class="message-body">
    <i class="fas fa-globe-americas mr-2"></i>本文同时提供以下语言的翻译：<a href="/hexo-theme-icarus/uncategorized/custom-hexo-tag-helpers/">English</a>。
  </div>
</article>
```

## 问题描述
个人感觉这种`html`的实现方式不是很好，写文章的时候不是很方便而且感觉不方便迁徙。想知道有没有一直类似`hexo tag`的写法，其他类似的语法如下所示：
```md
{% message color:<颜色> size:<大小> icon:<图标> title:<标题> %}
    <内容>
{% endmessage %}
```

<img width="754" alt="image" src="https://user-images.githubusercontent.com/77219045/219442605-b6693775-c083-4a7e-b225-b0ccd55f8c1e.png">

这样可以直接在写内容的地方写内容，我感觉在编写文章时候整体感觉会清晰很多。

**Comment by Mrzhang06** [comment, marked as answer]
刚才看了PPOffice大佬发布的文档源码[0132965](https://github.com/ppoffice/hexo-theme-icarus/blob/0132965ca3204761a2b0e89a29ec4979aa914cb6/source/_posts/zh-CN/Custom-Hexo-Tag-Helpers.md) ，以 raw 视图查看，实现方式也是上面那种HTML方式
